website/blog/2024-02-05-running-tabby-as-a-language-server/index.md

```diff
@@ -0,0 +1,85 @@
+    ![npx-run-tabby-agent](./npx-run-tabby-agent.gif)
+4. You can configure the agent's settings by editing the config file located at `~/.tabby-client/agent/config.toml`. If your Tabby server uses a different port or requires authentication, modify these settings accordingly:
+    ```toml
+    [server]
+    endpoint = "http://127.0.0.1:8080" # Replace with your server's endpoint
+    token = "your_token"
+    ```
+    For more details on configuration options, refer to [this documentation](https://tabby.tabbyml.com/docs/extensions/configurations).
+
+## Connect Your Editor to Tabby
+
+Most text editors support built-in LSP clients or popular LSP client plugins, making it easy to connect them to the Tabby agent language server. Let's take NeoVim and [coc.nvim](https://github.com/neoclide/coc.nvim) as an example to show you how to configure your editor to connect to Tabby.
+
+1. Install [coc.nvim](https://github.com/neoclide/coc.nvim) by following the [guide](https://github.com/neoclide/coc.nvim?tab=readme-ov-file#quick-start)
+2. Start NeoVim, and use the ``:CocConfig`` command to open the configuration file. Add the following configuration:
+    ```json
+    {
+      "languageserver": {
+        "tabby-agent": {
+          "command": "npx",
+          "args": ["tabby-agent", "--lsp", "--stdio"],
+          "filetypes": ["*"]
+        }
+      }
+    }
+    ```
+    The "filetypes": ["*"] setting enables Tabby for all filetypes. You can modify it according to your needs.
+
+3. Save the configuration file, and restart NeoVim. 
+4. Open a file and start typing code to see code completion suggestions from Tabby.
+
+![coc-tabby-completion](coc-tabby-completion.png)
+
+For more examples of connecting Tabby to other editors, refer to the [Tabby Agent documentation](https://github.com/tabbyml/tabby/tree/main/clients/tabby-agent/). If you have configurations for your favorite editors that you'd like to share, feel free to submit a pull request!
+
+## Create a Plugin for a New Editor
+
+In the previous examples, Tabby completions are displayed in the dropdown completion list. However, this method may not be very convenient for displaying multi-line code completions. As most LSP clients do not yet support [inline completion](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#textDocument_inlineCompletion), you may want to create a plugin for an editor that provides inline completion. To demonstrate how to communicate with Tabby via LSP, we have provided an example project [here](https://github.com/tabbyml/tabby/tree/main/clients/example-vscode-lsp).
+
+Please note that language server support is still in its early stages, and your feedback will be invaluable in making it even better. If you have any ideas or suggestions, feel free to create an issue or join our Slack community.
```

Comment:
Add a link to join Slack community.

Reply:
https://github.com/TabbyML/tabby/pull/1220/commits/102abb1be739e31b9c4e2fbed0d84e2fc7f44d2d